projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts

```diff
@@ -10,47 +10,96 @@ import {CONTEXT_TOKEN, LayerCollectionListComponent} from '../layer-collection-l
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class LayerCollectionNavigationComponent {
-    collections: Array<LayerCollectionItemDict | undefined> = [undefined];
+    collections: Array<LayerCollectionItemDict> = [];
+    allTrails: Array<Array<LayerCollectionItemDict>> = [];
+    displayedTrail: Array<LayerCollectionItemDict> = [];
 
-    selectedCollection = 0;
+    selectedCollection = -1;
 
     selectedPortal!: Portal<any>;
 
+    @ViewChild('scrollElement', {read: ElementRef}) public scrollElement!: ElementRef<any>;
+
     constructor() {
         this.setPortal(undefined);
     }
 
+    scrollToRight(): void {
+        setTimeout(() => {
+            // wait until breadcrumbs are re-rendered before scrolling
+            this.scrollElement.nativeElement.scrollLeft += Number.MAX_SAFE_INTEGER;
+        }, 0);
+    }
+
     selectCollection(id: LayerCollectionItemDict): void {
-        this.collections = this.collections.splice(0, this.selectedCollection + 1);
+        this.collections = this.collections.splice(0, this.displayedTrail.length);
         this.collections.push(id);
         this.selectedCollection += 1;
 
+        // Create a new trail, append it to the collection and display it
+        const clone = this.collections.map((x) => Object.assign({}, x));
+        this.allTrails = this.allTrails.slice(0, this.selectedCollection);
+        this.allTrails.push(clone);
+        this.displayedTrail = this.allTrails[this.selectedCollection];
+
+        this.scrollToRight();
+
         this.setPortal(id);
     }
 
     back(): void {
         if (this.selectedCollection > 0) {
             this.selectedCollection -= 1;
-            const id = this.collections[this.selectedCollection];
-
-            this.setPortal(id);
+            this.updateLayerView();
+        } else if (this.selectedCollection === 0) {
+            this.displayedTrail = [];
+            this.showRoot();
+            this.selectedCollection = -1;
         }
     }
 
     forward(): void {
-        if (this.selectedCollection < this.collections.length - 1) {
+        if (this.selectedCollection < this.allTrails.length - 1) {
             this.selectedCollection += 1;
-            const id = this.collections[this.selectedCollection];
-
-            this.setPortal(id);
+            this.updateLayerView();
+            this.scrollToRight();
         }
     }
 
+    updateLayerView(): void {
+        const currentTrail = this.allTrails[this.selectedCollection];
+        this.displayedTrail = currentTrail;
+        const lastId = currentTrail[currentTrail.length - 1];
+        this.setPortal(lastId);
+    }
+
+    onBreadCrumbClick(index: number): void {
+        // Creates and appends a new crumbtrail, then moves forward to it
+        if (index === this.displayedTrail.length - 1) return;
```

Comment:
please add curly braces around the return

Reply:
Added them in 0a65ceefb42d3a9b9bc603e0ea676a8a9eb358fb